chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -83,21 +83,29 @@ def get_nft_info_from_puzzle(puzzle: Program, nft_coin: Coin) -> NFTInfo:
     data_uris = []
     for uri in uncurried_nft.data_uris.as_python():
         data_uris.append(str(uri, "utf-8"))
+    meta_uris = []
+    for uri in uncurried_nft.meta_uris.as_python():
+        meta_uris.append(str(uri, "utf-8"))
+    license_uris = []
+    for uri in uncurried_nft.license_uris.as_python():
+        license_uris.append(str(uri, "utf-8"))
 
     nft_info = NFTInfo(
-        uncurried_nft.singleton_launcher_id.as_python().hex().upper(),
+        str(uncurried_nft.singleton_launcher_id).upper(),
         nft_coin.name().hex().upper(),
-        uncurried_nft.owner_did.as_python().hex().upper(),
+        str(uncurried_nft.owner_did).upper(),
         uint64(uncurried_nft.trade_price_percentage.as_int()),
         data_uris,
-        uncurried_nft.data_hash.as_python().hex().upper(),
-        [],
-        "",
-        [],
-        "",
-        "NFT0",
-        uint64(1),
-        uint64(1),
+        str(uncurried_nft.data_hash).upper(),
+        meta_uris,
+        str(uncurried_nft.meta_hash).upper(),
+        license_uris,
+        str(uncurried_nft.license_hash).upper(),
+        uint64(uncurried_nft.series_total.as_int()),
+        uint64(uncurried_nft.series_total.as_int()),
+        LAUNCHER_PUZZLE_HASH.hex().upper(),
+        str(uncurried_nft.metdata_updater_hash).upper(),
+        str(uncurried_nft.metadata).upper(),
```

Comment:
No uppercase conversion on this because it will cause problems with the case-sensitive comparison when creating an offer with the metadata. Also, the URLs in the metadata should not be modified.

```suggestion
        str(uncurried_nft.metadata),
```

Reply:
Ok, but the url will not be changed by this, since their are in hex.

Reply:
Oh I see

Reply:
Ok disregard if the data is still a hex string at this point.

Reply:
we should not be using hex strings in most places in the backed. You should probably just use bytes32. When you return something in the rpc_api, the middleware will convert everything to hexstr. Using strings internally will cause more bugs and hidden inconsistencies.

Reply:
1. This is only for the frontend, so bytes have to be serialized as hex in json.
2. I'm okay with lower case as long as we commit use it everywhere. The reason I use upper is because the upper case alignment is better than lower case. In lower case, even the hex length are same, they may look different.
For example,
dddddddddddddddddddd
ffffffffffffffffffff

Reply:
You don't need to serialize manually in the backend. Just retun bytes objects and the RPC server will translate everything to hex. It's bad practice to use hexstrings in the backend

Reply:
lower case vs upper case shoud not be an arguement, because we do not use hex in the backend.

Reply:
We have another data structure (UncurriedNFT) for backend usage. This NFTInfo is only used for frontend.

Reply:
non-JSON types are automatically serialized into json types when RPC serializes the data returned, hence no need to to do it in the data class

Reply:
I tried to switch to bytes/bytes32, the issue is the fields are serialized with "0x" prefix. The frontend may need to change. @paninaro I'm okay with any approach, what do you think?

Reply:
Sure, we can handle a "0x" prefix in the GUI